File: maximum_depth_of_binary_tree.cpp
/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int maxDepth(TreeNode *root)
    {
        if (root == NULL) return 0;
        int lmax = maxDepth(root->left);
        int rmax = maxDepth(root->right);
        return lmax > rmax ? lmax + 1 : rmax + 1;
    }
};

File: Python solutions/maximum_depth_of_binary_tree.py
# Definition for a  binary tree node
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    # @param root, a tree node
    # @return an integer

    def maxDepth(self, root):
        if root is None:
            return 0

        l_height = self.maxDepth(root.left)
        r_height = self.maxDepth(root.right)

        return max([l_height, r_height]) + 1
